glib/src/subclass/types.rs

```diff
@@ -95,7 +95,7 @@ unsafe impl<T: InstanceStruct> InstanceStructExt for T {
             let data = Self::Type::type_data();
             let private_offset = data.as_ref().impl_offset();
             let ptr: *const u8 = self as *const _ as *const u8;
-            let imp_ptr = ptr.offset(private_offset);
+            let imp_ptr = ((ptr as isize) + private_offset) as *const u8;
             let imp = imp_ptr as *const Self::Type;
```

Comment:
I think this is still wrong. There should be no `*const u8` anywhere here, and wherever it is used you should use `isize` or `usize`.